arbor/include/arbor/cable_cell_param.hpp

```diff
@@ -103,6 +104,98 @@ struct mechanism_desc {
     std::unordered_map<std::string, double> param_;
 };
 
+// FVM discretization policies/hints.
+
+class cable_cell;
+
+struct cv_policy_base {
+    virtual locset cv_boundary_points(const cable_cell& cell) const = 0;
+    virtual std::unique_ptr<cv_policy_base> clone() const = 0;
+    virtual ~cv_policy_base() {}
+};
+
+using cv_policy_base_ptr = std::unique_ptr<cv_policy_base>;
+
+struct cv_policy {
+    cv_policy(const cv_policy_base& ref) { // implicit
+        policy_ptr = ref.clone();
+    }
+
+    cv_policy(cv_policy&&) = default;
+
+    cv_policy(const cv_policy& other):
+        policy_ptr(other.policy_ptr->clone()) {}
+
+    cv_policy& operator=(const cv_policy&) = default;
+    cv_policy& operator=(cv_policy&&) = default;
+
+    locset cv_boundary_points(const cable_cell& cell) const {
+        return policy_ptr->cv_boundary_points(cell);
+    }
+
+private:
+    cv_policy_base_ptr policy_ptr;
+};
+
+// Common flags for CV policies; bitwise composable.
+namespace cv_policy_flag {
+    using value = unsigned;
+    enum : unsigned {
+        none = 0,
+        interior_forks = 1<<0,
+        single_root_cv = 1<<1
+    };
+}
+
+struct cv_policy_explicit: cv_policy_base {
```

Comment:
A short description of what each policy does would be good.

Reply:
Sure; I'll add some comments for policies and flags; I'll supply full docs later when the whole thing is done.